pkg/cloudprovider/providers/gce/gce_disks.go

```diff
@@ -299,28 +395,42 @@ func (gce *GCECloud) GetAutoLabelsForPD(name string, zone string) (map[string]st
 		}
 	}
 
-	region, err := GetGCERegion(zone)
-	if err != nil {
-		return nil, err
-	}
-
-	if zone == "" || region == "" {
+	labels := make(map[string]string)
+	switch zoneInfo := disk.ZoneInfo.(type) {
+	case singleZone:
+		if zoneInfo.zone == "" || disk.Region == "" {
+			// Unexpected, but sanity-check
+			return nil, fmt.Errorf("PD did not have zone/region information: %v", disk)
+		}
+		labels[kubeletapis.LabelZoneFailureDomain] = zoneInfo.zone
+		labels[kubeletapis.LabelZoneRegion] = disk.Region
+	case multiZone:
+		if zoneInfo.replicaZones == nil || zoneInfo.replicaZones.Len() <= 0 {
+			// Unexpected, but sanity-check
+			return nil, fmt.Errorf("PD is regional but does not have any replicaZones specified: %v", disk)
+		}
+		labels[kubeletapis.LabelZoneFailureDomain] = stringSetToCommaSeparatedString(zoneInfo.replicaZones)
```

Comment:
IIUC, the changes here will be used by the admission controller, and there's nothing to modify there.

Reply:
Correct, it just calls `GetAutoLabelsForPD(...)`, but this reminded me to update that call to handle a zone key with multiple values.